### PR TITLE
bootstrap ci for 3.0.x

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR breaths life back into the `v3.0.x` release feature branch.

It includes a hybrid collection of fixes from `master` and @rcomer to bootstrap the CI.

Note that, the recent GH action magic to auto-create `conda-lock` files is not fully implemented here. There is no GH action, as per `master`, but the same infra-structure is in-place but with statically created lock files.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/iris/src/developers_guide/contributing_code_formatting.rst [hunk 1]
```diff
@@ -39,7 +39,7 @@ linted according to our ``.flake8`` configuration file. Note that,
 for each ``.pre-commit-config.yaml`` git hook.
 
 Additionally, you may wish to enable ``black`` for your preferred
-`editor/IDE <https://black.readthedocs.io/en/stable/editor_integration.html#editor-integration>`_.
+`editor/IDE <https://black.readthedocs.io/en/stable/integrations/editors.html>`_.
```
Comment (bjlittle) [inline docs/iris/src/developers_guide/contributing_code_formatting.rst:42]: This was causing a `linkcheck` failure.

File: lib/iris/coord_categorisation.py [hunk 2]
```diff
@@ -103,14 +103,11 @@ def _pt_date(coord, time):
...
-        datetime.date
+        cftime.datetime
     """
     # NOTE: All of the currently defined categorisation functions are
     # calendar operations on Time coordinates.
-    #  - All these currently depend on Unit::num2date, which is deprecated (!!)
-    #  - We will want to do better, when we sort out our own Calendars.
-    #  - For now, just make sure these all call through this one function.
-    return coord.units.num2date(time)
+    return coord.units.num2date(time, only_use_cftime_datetimes=True)
```
Comment (bjlittle) [inline lib/iris/coord_categorisation.py:110]: Ported change, merge to my branch, thanks to @rcomer :smile:

File: lib/iris/fileformats/netcdf.py [hunk 3]
```diff
@@ -1043,7 +1043,7 @@ def write(
             dtype(i.e. 'i2', 'short', 'u4') or a dict of packing parameters as
             described below. This provides support for netCDF data packing as
             described in
-            http://www.unidata.ucar.edu/software/netcdf/docs/BestPractices.html#bp_Packed-Data-Values
+            http://www.unidata.ucar.edu/software/netcdf/documentation/NUG/best_practices.html#bp_Packed-Data_Values
```
Comment (lbdreyer) [inline lib/iris/fileformats/netcdf.py:1046]: A bit of an overlap with the changes in here: https://github.com/SciTools/iris/pull/4132/files, but we can rebase after we've merged

Comment (rcomer) [reply lib/iris/fileformats/netcdf.py:1046]: I think #4132 will be redundant once all tests are passing.

[the remark this answers is not in the document]
Comment (rcomer) [reply lib/iris/fileformats/netcdf.py:1046]:
```suggestion
            https://www.unidata.ucar.edu/software/netcdf/documentation/NUG/best_practices.html#bp_Packed-Data-Values
```

File: lib/iris/fileformats/netcdf.py [hunk 4]
```diff
@@ -2589,7 +2589,7 @@ def save(
         (i.e. 'i2', 'short', 'u4') or a dict of packing parameters as described
         below or an iterable of such types, strings, or dicts.
         This provides support for netCDF data packing as described in
-        http://www.unidata.ucar.edu/software/netcdf/docs/BestPractices.html#bp_Packed-Data-Values
+        http://www.unidata.ucar.edu/software/netcdf/documentation/NUG/best_practices.html#bp_Packed-Data_Values
```
Comment (rcomer) [inline lib/iris/fileformats/netcdf.py:2592]:
```suggestion
        https://www.unidata.ucar.edu/software/netcdf/documentation/NUG/best_practices.html#bp_Packed-Data-Values
```

**Review by rcomer — COMMENTED** [review]
Suggested link change got tests passing at https://github.com/bjlittle/iris/pull/33

**Review by rcomer — COMMENTED** [review]
Suggested link change got tests passing at https://github.com/bjlittle/iris/pull/33

File: .cirrus.yml [hunk 5]
```diff
@@ -52,54 +52,75 @@ env:
...
 
 
 #
-# YAML alias for compute credits
+# YAML alias for compute credits.
 #
 compute_credits_template: &CREDITS_TEMPLATE
   # Only use credits for non-DRAFT pull-requests to SciTools/iris master branch by collaborators
-  use_compute_credits: $CIRRUS_REPO_FULL_NAME == 'SciTools/iris' && $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR_DRAFT == 'false' && $CIRRUS_BASE_BRANCH == 'master' && $CIRRUS_PR != ''
+  use_compute_credits: ${CIRRUS_REPO_FULL_NAME} == "SciTools/iris" && ${CIRRUS_USER_COLLABORATOR} == "true" && ${CIRRUS_PR_DRAFT} == "false" && ${CIRRUS_PR} != ""
```
Comment (lbdreyer) [inline .cirrus.yml:91]: Perhaps change the comment on line 90 to no longer specifiy "master"?

**Comment by bjlittle** [comment]
@lbdreyer Brilliant. Thanks :smile:

**Pull request merged**